pkg/build/util/generate_test.go

```diff
@@ -352,3 +532,22 @@ func mockBuildConfig(source api.BuildSource, strategy api.BuildStrategy, output
 		},
 	}
 }
+
+type mockImageRepositoryNamespaceGetter struct {
+	namespace string
+	name      string
+}
+
+func (m *mockImageRepositoryNamespaceGetter) GetByNamespace(namespace, name string) (*imageapi.ImageRepository, error) {
+	if m.namespace == namespace && m.name == name {
+		return &imageapi.ImageRepository{
+			ObjectMeta: kapi.ObjectMeta{Name: imageRepoName},
+			Status: imageapi.ImageRepositoryStatus{
+				DockerImageRepository: originalImage,
+			},
+			Tags: map[string]string{tagName: newTag},
+		}, nil
+	} else {
```

Comment:
For reference, here's the convention we're following: https://golang.org/doc/effective_go.html#if

----- Original Message -----
> > +
> > +type mockImageRepositoryNamespaceGetter struct {
> > +	namespace string
> > +	name      string
> > +}
> > +
> > +func (m *mockImageRepositoryNamespaceGetter) GetByNamespace(namespace,
> > name string) (*imageapi.ImageRepository, error) {
> > +	if m.namespace == namespace && m.name == name {
> > +		return &imageapi.ImageRepository{
> > +			ObjectMeta: kapi.ObjectMeta{Name: imageRepoName},
> > +			Status: imageapi.ImageRepositoryStatus{
> > +				DockerImageRepository: originalImage,
> > +			},
> > +			Tags: map[string]string{tagName: newTag},
> > +		}, nil
> > +	} else {
> 
> the current form is more readable than the inverted logic, imho.  I would
> argue that fewer negations in boolean logic is more valuable than fewer
> indentations in code blocks, though i'm sure there are extreme examples for
> either case.
> 
> also i dislike the practice of not being explicit in having an else block.
> it means the future reader needs to have noticed there is a return in the
> if-block to understand whether or not the non-elsed logic will execute.
> it's practically a goto at that point.  I can see cases where the
> additional else block could reduce readability, but this doesn't seem like
> one of those.
> 
> 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/889/files#r24356553